lib/mqtt/MqttController.js

```diff
@@ -0,0 +1,248 @@
+     */
+    constructor(options) {
+        this.loadConfig();
+
+        this.client = null;
+        // Workaround to be able to set two Last Will and Testament payloads, one for Homie and one for HAss
+        // this.clientSecondaryLWT = null; // TODO actually implement
+
+        /** @type {import(./NodeMqttHandler)} */
+        this.nodes = [];
+
+        /**
+         * @callback subscribeCb
+         * @param {string} value
+         * @return {Promise<void>}
+         */
+        /** @type {object.<subscribeCb>} */
+        this.subscriptions = {};
+
+        this.initialized = false;
+
+        this.loadConfig();
+
+        this.robotHandle = new RobotMqttHandle({
+            robot: options.robot,
+            controller: this,
+            baseTopic: this.topicPrefix,
+            topicName: this.identifier,
+            friendlyName: this.friendlyName
+        });
+
+        if (this.enabled) {
+            this.connect().then();
+        }
+
+        this.config.onUpdate(async (key) => {
+            if (key === "mqtt") {
+                await this.reconfigure(async () => {
+                    await this.robotHandle.deconfigure();
+                });
+
+                await this.disconnect();
+                this.loadConfig();
+
+                this.robotHandle = new RobotMqttHandle({
+                    robot: options.robot,
+                    controller: this,
+                    baseTopic: this.topicPrefix,
+                    topicName: this.identifier,
+                    friendlyName: this.friendlyName
+                });
+
+                if (this.enabled) {
+                    await this.connect();
+                }
+            }
+        });
+    }
+
+    /**
+     * @private
+     */
+    loadConfig() {
+        const mqttConfig = this.config.get("mqtt");
+
+        this.enabled = mqttConfig.enabled;
+        this.server = mqttConfig.server;
+        this.clientId = mqttConfig.clientId;
+        this.topicPrefix = mqttConfig.topicPrefix;
+        this.port = mqttConfig.port ?? 1883;
+        this.identifier = mqttConfig.identifier;
+        this.friendlyName = mqttConfig.friendlyName;
+        this.username = mqttConfig.username;
+        this.password = mqttConfig.password;
+        this.usetls = mqttConfig.usetls;
+        this.ca = mqttConfig.ca ?? "";
+        this.clientCert = mqttConfig.clientCert ?? "";
+        this.clientKey = mqttConfig.clientKey ?? "";
+        this.qos = mqttConfig.qos ?? 0;
+
+        this.homieEnabled = mqttConfig.homie.enabled;
+
+        this.attributesUpdateInterval = mqttConfig.attributesUpdateInterval ?? 60000;
+        this.provideMapData = mqttConfig.provideMapData !== undefined ? mqttConfig.provideMapData : true;
+
+        this.hassEnabled = mqttConfig.hass.enabled; // TODO actually implement hass
+        //this.homeassistantMapHack = mqttConfig.homeassistant.mapHack !== undefined ? mqttConfig.homeassistant.mapHack : true; // TODO
```

Comment:
We can drop the Map Hack option and make it on by default with this rewrite, because non-HA users can simply use Homie then and therefore there's no point in being able to turn that off.

Reply:
Alright. In general all the commented hass stuff is just there as a reminder to take care of them later, I haven't seen yet what the hack does.